src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -21,27 +22,28 @@ public class LibGpiodDriver : UnixDriver
 
         public LibGpiodDriver()
         {
-            SafeChipIteratorHandle iterator = null;
-            try
+            // set /dev/gpiochip0 as default if hardware has only one GPIO bank
+            if (Interop.GetNumberOfChips() == 1)
```

Comment:
Some Linux version expose more than one chips even there were only one gpio chip. In that case by this logic any chip not loaded until below SetDefaultGpioChip(..) method called from controller.
What i would prefer is load the chip on driver constructor only (i don't think we would need to change a driver or chip for a controller, if we want to access another chip create another controller with a driver pointing to that chip). Actually all other driver methods written for only one same chip usage, chip shouldn't be changed after driver creation. So chip number should be set with driver constructor once, shouldn't be updatable further

Reply:
Basically something like this
```
public LibGpiodDriver(int chipId = 0) {
    try
    {
        _chip = Interop.OpenChipByNumber(gpioChip);
        if (_chip == null)
        {
            throw ExceptionHelper.GetIOException(ExceptionResource.NoChipFound, Marshal.GetLastWin32Error());
        }
        _pinNumberToSafeLineHandle = new Dictionary<int, SafeLineHandle>(PinCount);
    } 
    catch (DllNotFoundException)
    {
        throw ExceptionHelper.GetPlatformNotSupportedException(ExceptionResource.LibGpiodNotInstalled);
     }
 }
}
```

Reply:
Thanks @buyaa-n I put it in 940bc17

Reply:
Thank you @microhobby